sui-execution/latest/sui-adapter/src/gas_charger.rs

```diff
@@ -244,6 +245,15 @@ pub mod checked {
             self.gas_status.charge_storage_read(total_size)
         }
 
+        pub fn charge_regulated_transfers(
+            &mut self,
+            regulated_coin_types_checked: u64,
+        ) -> Result<(), ExecutionError> {
+            let cost_per_type = CONFIG_READ_BYTES_SIZE_ESTIMATE * 2;
+            self.gas_status
+                .charge_storage_read(cost_per_type * (regulated_coin_types_checked as usize))
```

Comment:
Right now I'm just scaling it by a factor of 2 for each config setting. Do we want any additional fudge factoring here to try to account for the hashing work being done?